app/kumactl/cmd/delete/delete_meshes_test.go

```diff
@@ -0,0 +1,165 @@
+		{
+			Spec: v1alpha1.Mesh{
+				Mtls: &v1alpha1.Mesh_Mtls{
+					Enabled: true,
+					Ca: &v1alpha1.CertificateAuthority{
+						Type: &v1alpha1.CertificateAuthority_Builtin_{
+							Builtin: &v1alpha1.CertificateAuthority_Builtin{},
+						},
+					},
+				},
+				Logging: &v1alpha1.Logging{
+					AccessLogs: &v1alpha1.Logging_AccessLogs{
+						Enabled:  true,
+						FilePath: "/tmp/access.log",
+					},
+				},
+			},
+			Meta: &test_model.ResourceMeta{
+				Mesh:      "mesh1",
+				Name:      "mesh1",
+				Namespace: "default",
+			},
+		},
+		{
+			Spec: v1alpha1.Mesh{
+				Mtls: &v1alpha1.Mesh_Mtls{
+					Enabled: false,
+					Ca: &v1alpha1.CertificateAuthority{
+						Type: &v1alpha1.CertificateAuthority_Builtin_{
+							Builtin: &v1alpha1.CertificateAuthority_Builtin{},
+						},
+					},
+				},
+			},
+			Meta: &test_model.ResourceMeta{
+				Mesh:      "mesh2",
+				Name:      "mesh2",
+				Namespace: "default",
+			},
+		},
+	}
+
+	Describe("DeleteMeshCommand", func() {
+
+		var rootCtx *kumactl_cmd.RootContext
+		var rootCmd *cobra.Command
+		var outbuf, errbuf *bytes.Buffer
+		var store core_store.ResourceStore
+
+		BeforeEach(func() {
+			// setup
+			rootCtx = &kumactl_cmd.RootContext{
+				Runtime: kumactl_cmd.RootRuntime{
+					Now: func() time.Time { return time.Now() },
+					NewResourceStore: func(*config_proto.ControlPlaneCoordinates_ApiServer) (core_store.ResourceStore, error) {
+						return store, nil
+					},
+				},
+			}
+
+			store = memory_resources.NewStore()
+
+			for _, ds := range sampleMeshes {
+				key := core_model.MetaToResourceKey(ds.Meta)
+				err := store.Create(context.Background(), ds, core_store.CreateBy(key))
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			rootCmd = cmd.NewRootCmd(rootCtx)
+			outbuf = &bytes.Buffer{}
+			errbuf = &bytes.Buffer{}
+			rootCmd.SetOut(outbuf)
+			rootCmd.SetErr(errbuf)
+		})
+
+		It("should throw an error in case of no args", func() {
+			// given
+			rootCmd.SetArgs([]string{
+				"--config-file", filepath.Join("..", "testdata", "sample-kumactl.config.yaml"),
+				"delete", "mesh"})
+
+			// when
+			err := rootCmd.Execute()
+
+			// then
+			Expect(err).To(HaveOccurred())
+			// and
+			Expect(err.Error()).To(Equal("accepts 1 arg(s), received 0"))
+			// and
+			Expect(outbuf.String()).To(MatchRegexp(`Error: accepts 1 arg\(s\), received 0`))
+			// and
+			Expect(errbuf.Bytes()).To(BeEmpty())
+		})
+
+		It("should throw an error in case of a non existing mesh", func() {
+			// given
+			rootCmd.SetArgs([]string{
+				"--config-file", filepath.Join("..", "testdata", "sample-kumactl.config.yaml"),
+				"delete", "mesh", "some-non-existing-mesh"})
+
+			// when
+			err := rootCmd.Execute()
+
+			// then
+			Expect(err).To(HaveOccurred())
+			// and
+			Expect(err.Error()).To(MatchRegexp("failed to get Mesh with the name some-non-existing-mesh: .*"))
+			// and
+			Expect(outbuf.String()).To(MatchRegexp("failed to get Mesh with the name some-non-existing-mesh: .*"))
```

Comment:
Let's check that the error is exactly `Error: there is no Mesh with name "some-non-existing-mesh"`